### ISSUE TITLE
used_underscore_binding triggered on #[derive(RustcEncodable)] and macros

### ISSUE THREAD

**Issue by karyon** [Manishearth/rust-clippy#536]
in servo, i get the used_underscore_binding lint about 2000 times. most of the times, it's on usages of #[derive(RustcEncodable)] and in code generated by the bitflags! macro.

**Comment by llogiq** [comment]
Another macro check then?

**Comment by Manishearth** [comment]
This was fixed, again not in the Servo branch https://github.com/Manishearth/rust-clippy/commit/b6766a0dcf66650a24820929c20be237e37230bc

**Comment by karyon** [comment]
dang, i actually updated that branch as far as possible, but had to stop before one of the rustups, 4 commits before the commit you mentioned. @Manishearth you think the bitflags macro is fixed as well?

**Comment by Manishearth** [comment]
It should be.

I'm landing a rustup in Servo soon so that we can get the latest clippy.

**Comment by Luthaf** [comment]
This warning is still triggered on `bitflags!` macro with the latest master. Should I do something special to remove it ?

**Comment by Manishearth** [comment]
Can you give an example of a bitflags invocation that causes this? I can't reproduce this locally.

**Comment by Luthaf** [comment]
It is this line: https://github.com/Luthaf/cymbalum/blob/c0baf13d6ee9257aaa4490192fb6532741ee3e1c/src/universe/molecules.rs#L113, which produces the following output on my computer:

```
<bitflags macros>:18:6: 18:12 warning: used binding which is prefixed with an underscore. A leading underscore signals that a binding will not be used., #[warn(used_underscore_binding)] on by default
<bitflags macros>:18 if ! _first { try ! ( f . write_str ( " | " ) ) ; } _first = false ; try ! (
                          ^~~~~~
<bitflags macros>:90:1: 92:63 note: in this expansion of bitflags! (defined in <bitflags macros>)
src/universe/molecules.rs:113:5: 127:6 note: in this expansion of bitflags! (defined in <bitflags macros>)
<bitflags macros>:18:6: 18:12 warning: used binding which is prefixed with an underscore. A leading underscore signals that a binding will not be used., #[warn(used_underscore_binding)] on by default
<bitflags macros>:18 if ! _first { try ! ( f . write_str ( " | " ) ) ; } _first = false ; try ! (
                          ^~~~~~
<bitflags macros>:90:1: 92:63 note: in this expansion of bitflags! (defined in <bitflags macros>)
src/universe/molecules.rs:113:5: 127:6 note: in this expansion of bitflags! (defined in <bitflags macros>)
<bitflags macros>:18:6: 18:12 warning: used binding which is prefixed with an underscore. A leading underscore signals that a binding will not be used., #[warn(used_underscore_binding)] on by default
<bitflags macros>:18 if ! _first { try ! ( f . write_str ( " | " ) ) ; } _first = false ; try ! (
                          ^~~~~~
<bitflags macros>:90:1: 92:63 note: in this expansion of bitflags! (defined in <bitflags macros>)
src/universe/molecules.rs:113:5: 127:6 note: in this expansion of bitflags! (defined in <bitflags macros>)
<bitflags macros>:18:6: 18:12 warning: used binding which is prefixed with an underscore. A leading underscore signals that a binding will not be used., #[warn(used_underscore_binding)] on by default
<bitflags macros>:18 if ! _first { try ! ( f . write_str ( " | " ) ) ; } _first = false ; try ! (
                          ^~~~~~
<bitflags macros>:90:1: 92:63 note: in this expansion of bitflags! (defined in <bitflags macros>)
src/universe/molecules.rs:113:5: 127:6 note: in this expansion of bitflags! (defined in <bitflags macros>)
```

I tried to reproduce in a minimal example but I could not. It may be some kind of interaction with private modules and re-exporting stuff everywhere.

Additional information:

```
$ rustc --version
rustc 1.8.0-nightly (0486e12ad 2016-01-23)
$ cargo build --features lint
  ...
  Compiling clippy v0.0.37 (https://github.com/Manishearth/rust-clippy/#5f17b38b)
  ...
```

**Comment by Manishearth** [comment]
Update your bitflags crate, this was fixed. It's a bug in the `Debug` impl

**Comment by Manishearth** [comment]
http://github.com/rust-lang-nursery/bitflags/commit/9226d14be6e980d42f431342710b03c8ab720473

**Comment by Luthaf** [comment]
That was it, thank you!
